components/scream/src/share/atm_process/atmosphere_process_group.cpp

```diff
@@ -144,25 +144,174 @@ void AtmosphereProcessGroup::finalize_impl (/* what inputs? */) {
 }
 
 void AtmosphereProcessGroup::
-set_required_group_impl (const FieldGroup<const Real>& group)
-{
+set_required_field (const Field<const Real>& f) {
+  if (m_group_schedule_type==ScheduleType::Parallel) {
+    // In parallel splitting, all required fields are *actual* inputs,
+    // and the base class impl is fine.
+    AtmosphereProcess::set_required_field(f);
+  }
+
```

Comment:
Thank you for adding the check, could we add another check that the ScheduleType is Sequential and throw an error if not?  I'm thinking that in the future we may implement other ScheduleTypes we've never heard of and putting in as many hooks as possible that make sure we don't accidentally break the code without a map to find out where would be great.

Reply:
Do you really think there will be a schedule type other than Sequential or Parallel?

Reply:
none specifically come to mind.  I guess from one perspective, anything we come up with will just be a combination of those two elemental coupling approaches.  I suppose if we decide to do something exotic it will be important that whatever that is be not a new ScheduleType per-se but some set of operations that are Sequential or Parallel.

Reply:
Well, the way I see it, a "mixed" approach will be likely consist of sub-groups, so each individual sub-group will still be sequential or parallel.